docker-compose.yaml

```diff
@@ -0,0 +1,23 @@
+version: '3.6'
+services:
+  bsc:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    command: ["--http.addr", "0.0.0.0", "--ws.addr", "0.0.0.0"]
+    environment:
+      - LOG_LEVEL=3
+      - DIFF_SYNC=true
```

Comment:
Most of the flags do not yet support the reading of env...
Must use additional script to convert to flag

Reply:
the reason which I suggest to only support 2 BSC unique env: `BSC_NETWORK` & `BSC_DATA_DIR` is that `geth` has quite a few command line options and is quite familiar by the developers. to keep the compatibility and minimise the maintenance for the script. `BSC_NETWORK` is used to boot the node to different network. default `/data` will be convenient to configure docker image, specially when hosting in k8s when mounting volume. `config.toml` & `genesis.json` can also be mounted by `configmap` in k8s into the `BSC_HOME/mainnet/` or`BSC_HOME/testnet/` before bootstrap script kicks in.

Reply:
another thought on building network specific image is to use another `ARG: <mainnet|testnet>` to build image for `mainnet` & `testnet` separately, this will end up with two image tags, e.g. `ghcr.io/bnb-chain/bsc:v1.1.11-mainnet`, `ghcr.io/bnb-chain/bsc:v1.1.11-testnet`. the config in the docker image will only host the config from the network..this will give much clean image but extra image tag.

Reply:
1. I agree with you, but I just want to say that the `DIFF_SYNC=true` environment variable has no effect here
2. I think building it as an image is enough, the program is the same, just the configuration is different

Reply:
sample docker-compose file just updated to reflect the usage with command line.
in k8s, it will look like
```
     containers:
      - name: client
        image: bnb-chain/bsc:v1.1.11
        args: ["--http.addr", "0.0.0.0", "--http.port", "8545", "--http.vhosts", "*", "--verbosity", "3"]
        env:
          - name: BSC_NETWORK
            value: mainnet
```